File: Models/Lesson.cs
using System;
using System.Collections.Generic;

namespace Simp.Models
{
    public class Lesson
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public LessonType LessonType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public Uri MeetingUri { get; set; }
        
        public Classroom Classroom { get; set; }
        public Venue Venue { get; set; }
        public ICollection<ApplicationUser> Teachers { get; set; }
        public ICollection<ApplicationUser> Attendees { get; set; }
    }
}

File: ClientApp/src/models/Lesson.ts
import User from "./User";

import LessonType from "./LessonType";

export interface LessonInstance {
    Id: string;
    Name: string;
    Description: string;
    LessonType: LessonType;
    StartDate: number;
    EndDate: number;
    MeetingUri: string;
    Teachers: User[];
}

export default LessonInstance;